js/components/Confirmation.jsx

```diff
@@ -5,21 +5,219 @@ import ProgressButton from "us-forms-system/lib/js/components/ProgressButton";
+
+            request.send(JSON.stringify(formData));
+            })
+            .then(function (response) {
+                formContext.markDone(formContext);
+                formContext.clearLocalStorage();
+                return response;
+            })
+            .catch(function (error) {
+                console.log("Error: ");
+                console.log(error);
+                // return error;
+            });
+    }
+
+    // Renders the right button based on the state formSubmissionState
+    // 0 = Submit (regular)
+    // 1 = Loading (three dots animation)
+    // 2 = Done (animated checkmark, green button)
+    renderButton() {
+        switch(this.state.formSubmissionState) {
+            case 1:
+                return <button className="confirmation-button" onClick={() => this.resubmitForm()}>
+                    <div className="confirmation-button--loading-btn"></div>
+                </button>;
+            case 2:
+                return <button className="confirmation-button confirmation-button__success" onClick={null}>
+                    <div className="confirmation-button--checkmark-container">
+                        <div className="confirmation-button--checkmark draw">&nbsp;</div>
+                    </div>
+                </button>;
+            default:
+                return <button className="confirmation-button" onClick={() => this.resubmitForm()}>Submit</button>;
+        }
+    }
+
+    render() {
+        let confirmationCaseNumber = this.state.confirmationCaseNumber;
+        let confirmation_content = null;
+
+        if(this.state.formSubmissionState === -1) {
+            console.log("renderConfirmation() The email looks valid, so no need to render form");
+            confirmation_content = <div style={{marginTop: "4rem"}}>
+                You will receive an email with a copy of your complaint and a confirmation number at <b>{this.state.userEmail}</b>
+            </div>;
+
+            this.clearLocalStorage();
+        } else {
+            let buttonContent = this.renderButton();
+            confirmation_content = <div className="confirmation">
+                <p style={{fontSize: "1.4rem"}}>For a copy of your complaint and confirmation number, enter your email address below.</p>
+                <span style={{fontSize: "2rem"}}>Your email</span>
+                <input className="confimation-input" type="email" disabled = {(this.state.formSubmissionState > 0) ? "disabled" : ""} value={this.state.userEmail} onChange={(event) => this.setState({userEmail: event.target.value})} />
+                {buttonContent}
+                <div className="confirmation-errorbox" style={(this.state.errorMessage.length === 0) ? {display: "none"} : {display: "block"}}>{this.state.errorMessage} (<a href="javascript:;" onClick={() => this.hideErrorMessage()}>dismiss</a>)</div>
+            </div>;
+        }
+
+        return (
+          <div className="schemaform-intro">
```

Comment:
I know we've been inconsistent, but we have some inline styles but also have styles added in overrides.css. We should probably refactor these inline styles into the stylesheet, some of them may even be redundant. I'd be fine with waiting til after press push to do this, though.